cmd/kogito/command/install/data_index.go

```diff
@@ -239,7 +210,8 @@ func (i *installDataIndexCommand) Exec(cmd *cobra.Command, args []string) error
 
 	return shared.
 		ServicesInstallationBuilder(i.Client, i.flags.Project).
-		OperatorInstalled().
+		SilentlyInstallOperatorIfNotExists().
+		WarnIfDependenciesNotReady(i.flags.infinispan.UseKogitoInfra, i.flags.kafka.UseKogitoInfra).
 		InstallDataIndex(&kogitoDataIndex).
```

Comment:
There is a different behavior here between the `kogito new|use-project --install-XXX` command and the `kogito install XXX` command.
Indeed, into the `kogito install XXX` command, the default image is set to `infrastructure.DefaultDataIndexImageNoVersion+infrastructure.GetRuntimeImageVersion()` where as in the `kogito new|use-project --install-XXX`, there is none by default.

So if I use another CLI version than the operator then the installation, with `kogito install XXX` command, it tries to install the version from the CLI.
For example in my tests CLI is version 0.10 and OperatorHub is version 0.9.1. So executing `kogito install data-index` with CLI on version 0.10, it tries to get the data-index version 0.10, which does not exist yet.
But this could be in another way, if user did not update to latest CLI, for example still using version 0.8.0, then it would install version 0.8 instead of 0.9 expected.

IMO, we should let the [image flag](https://github.com/kiegroup/kogito-cloud-operator/blob/f469c6f5f738fdadde943624a7d8a8a2464a4817/cmd/kogito/command/install/data_index.go#L143) as empty string by default, instead of `infrastructure.DefaultDataIndexImageNoVersion+infrastructure.GetRuntimeImageVersion()`. Then the image will be set by the operator itself and not the CLI.
And this of course for data-index, jobs-service and management-console.

wdyt ?